src/theme/NavbarItem/NavbarNavLink.tsx

```diff
@@ -24,25 +31,51 @@ export default function NavbarNavLink({
   const normalizedHref = useBaseUrl(href, { forcePrependBaseUrl: true });
   const isExternalLink = label && href && !isInternalUrl(href);
 
-  const [data, setData] = useState(null);
+  const [githubStars, setGithubStars] = useState(null);
 
   useEffect(() => {
     const getData = async () => {
+      const cachedGithubStars = JSON.parse(
+        sessionStorage.getItem(GITHUB_STARS_SESSION_STORAGE_NAME),
+      ) as unknown as githubStarsSessionStorage;
       try {
+        if (cachedGithubStars) {
+          const cacheExpiryTime = new Date(cachedGithubStars.retrievedTime);
+          cacheExpiryTime.setDate(cacheExpiryTime.getDate() + 1)
+          if ( cacheExpiryTime.getTime() > Date.now()) {
+            setGithubStars(cachedGithubStars.count);
+            return;
+          }
+        }
+
+        // cache should have been expired.  Let's fetch more recent data
         const response = await fetch(`https://api.github.com/repos/openfga/openfga`);
         if (!response.ok) {
           throw new Error(`This is an HTTP error: The status is ${response.status}`);
         }
         const actualData = await response.json();
-        setData(actualData.stargazers_count);
+        setGithubStars(actualData.stargazers_count);
```

Comment:
Have this below the next section, and just do:
```suggestion
        setGithubStars(newCachedGithubStars);
```